akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -153,6 +154,20 @@ private[akka] object Shard {
 
   case object PassivateIdleTick extends NoSerializationVerificationNeeded
 
+  private[sharding] def name(id: ShardRegion.ShardId): String =
+    URLEncoder.encode(id, ByteString.UTF_8)
+
+  /**
+   * INTERNAL API
+   *
+   * Use to determine whether an `ActorRef` is a shard.
+   * Returns true if the `path` is for a shard, meaning the name was encoded from the `shardId`.
+   * Region and region proxy names are encoded from their `typeName`, the coordinator name is derived from the encoded region name.
+   */
+  private[sharding] def isShard(typeName: String, shardId: ShardRegion.ShardId, path: ActorPath): Boolean =
+    path.parent.name.contains(typeName) && // region or proxy
+    path.name == shardId
```

Comment:
This function is why I chose to consolidate the 3 namings, I wanted the shards and once you do that, you want it  consistent.